### PR TITLE
Clh driver: removed hard-coded vsock contextid (cid)

### PR DESCRIPTION
**Pull request by ericooper**
Removed hardcoded cid in the driver and replaced it by a call to utils find context id. To support this approach an addition of a contextid field in the hybrid vsock type was necessary in sandbox

### CODE REVIEW

File: virtcontainers/clh.go [hunk 1]
```diff
@@ -436,7 +429,13 @@ func (clh *cloudHypervisor) addDevice(devInfo interface{}, devType deviceType) e
 			device: v.Name(),
 			mac:    v.HardwareAddr(),
 		})
-
+	case types.HybridVSock:
+		clh.Logger().WithField("function", "addDevice").Infof("Adding HybridVSock: path=%s, cid=%d, port=%d", v.UdsPath, v.ContextID, v.Port)
```
Comment (jodh-intel) [inline virtcontainers/clh.go]: Could you change the log calls to be "structured" by pulling out the relevant info into a set of fields:

```go
.WithFields(logrus.Fields{
    "function" : "addDevice",
    "path" : v.UdsPath,
    "cid" : v.ContextID,
    "port" : v.Port,
}).Info("Adding HybridVSock")
```

File: virtcontainers/clh.go [hunk 2]
```diff
@@ -544,19 +543,23 @@ func (clh *cloudHypervisor) reset() {
...
 
 	udsPath, err := clh.vsockSocketPath(id)
 	if err != nil {
 		clh.Logger().Info("Can't generate socket path for cloud-hypervisor")
 		return types.HybridVSock{}, err
 	}
-	clh.Logger().WithField("function", "generateSocket").Infof("Using hybrid vsock %s:%d", udsPath, vSockPort)
+	_, cid, err := utils.FindContextID()
+	if err != nil {
+		return nil, fmt.Errorf("Can't generate hybrid vsocket for cloud-hypervisor: error finding available context id")
```
Comment (jodh-intel) [inline virtcontainers/clh.go]: Please include `err` in the returned error.

File: virtcontainers/clh.go (hunk 1 above)
Comment (ericooper) [reply virtcontainers/clh.go]: Done.

File: virtcontainers/clh.go (hunk 2 above)
Comment (ericooper) [reply virtcontainers/clh.go]: Done

**Comment by ericooper** [comment]
> Thanks @ericooper.
> 
> lgtm but please reformat the file using `gofmt -s` as shown in the Travis log:
> 
> ```
> virtcontainers/types/sandbox.go:190: File is not `gofmt`-ed with `-s` (gofmt)
> 	UdsPath string
> 	ContextID uint64	
> 	Port    uint32
> The command ".ci/static-checks.sh" failed and exited with 1 during .
> ```
> 
> Note that you can run the [`static-checks.sh`](https://github.com/kata-containers/tests/blob/master/.ci/README.md) script locally to avoid CI surprises once you push smile

Sure. I didn't know, thanks for the tip. I've applied the ci script and did an update so it should be good to go now. Cheers.

**Pull request merged**